### PR TITLE
refactor(react-components): remove casting by filtering and move the clean content to a function

### PR DESCRIPTION
**Pull request by danpriori**
#### Type of change
Refactor 

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:
Some small refactoring to reduce the ugliness of the rush:
Remove a casting by returning only contents with strings and move the content cleanup logic into a separated function. 

## Checklist :ballot_box_with_check:

- [ ] I am proud of this feature.
- [X] I have performed a self-review of my own code.
- [X] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [X] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/architecture/concrete/pointsOfInterest/CreatePointsOfInterestWithDescriptionCommand.tsx [hunk 1]
```diff
@@ -110,3 +113,18 @@ export class CreatePointsOfInterestWithDescriptionCommand extends CustomBaseInpu
...
+      if (
+        content.type === 'text' ||
+        content.type === 'comment' ||
+        content.type === 'commentWithButtons'
+      ) {
+        return content.content;
+      }
+      return undefined;
+    })
+    .filter((content) => !isUndefined(content));
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/concrete/pointsOfInterest/CreatePointsOfInterestWithDescriptionCommand.tsx:129]: I think we have a `isDefined` in react-components - so you could write `.filter(isDefined)`

File: react-components/src/architecture/concrete/pointsOfInterest/types.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export type PointOfInterest<IdType> = {
 export function createPointsOfInterestPropertiesFromPointAndTitle(
   point: Vector3,
   scene: DmsUniqueIdentifier,
-  contents: ReactNode[],
+  contents: string[],
   associatedInstance: InstanceReference | undefined
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/concrete/pointsOfInterest/types.ts:31]: I think want this function to take the `title` + `description` explicitly in their own parameters. Right now it's a bit cryptic what the `contents`-argument means :sweat_smile:

**Review by haakonflatval-cognite — APPROVED** [review]
A couple of suggestions, otherwise looks good :+1:

File: react-components/src/architecture/concrete/pointsOfInterest/types.ts (hunk 2 above)
Comment (danpriori) [reply react-components/src/architecture/concrete/pointsOfInterest/types.ts]: yes sure
https://github.com/cognitedata/reveal/pull/4929/commits/c0bebe9a063de9f6d742f5648af6bf4636363e34

File: react-components/src/architecture/concrete/pointsOfInterest/CreatePointsOfInterestWithDescriptionCommand.tsx (hunk 1 above)
Comment (danpriori) [reply react-components/src/architecture/concrete/pointsOfInterest/CreatePointsOfInterestWithDescriptionCommand.tsx]: ohh cool
done
https://github.com/cognitedata/reveal/pull/4929/commits/c0bebe9a063de9f6d742f5648af6bf4636363e34

**Pull request merged**